### ISSUE TITLE
Check the minimum version of dune during opam file generation

### ISSUE THREAD

**Issue by kit-ty-kate** [ocaml/dune#3427]
When turning on the opam file generation, if the dune dependency is explicitly specified no check is being done on whether or not the dune dependency actually has a correct constraint.
For instance:
```
$ ls
$ cat dune-project
(lang dune 2.4)
(name test)

(generate_opam_files true)

(package
  (name test)
  (depends dune))
$ dune build
$ cat test.opam
# This file is generated by dune, edit dune-project instead
opam-version: "2.0"
depends: ["dune"]
[...]
```
Here the opam file is generated with an explicit dependency to dune with no constraint. In my opinion it should be checked that `>= 2.4` or above is being required. It would simplify opam-repository maintenance. For example see this PR in which the case was hit: https://github.com/ocaml/opam-repository/pull/16280

**Comment by kit-ty-kate** [comment]
Second PR in a row with the same issue today: https://github.com/ocaml/opam-repository/pull/16281

**Comment by jeremiedimino** [comment]
Indeed. If the dependency on dune is omitted, we automatically insert one with the right constraint. But if one is already present, we don't add the constraint.

**Comment by avsm** [comment]
Makes sense to add the constraint unconditionally indeed -- there's no situation that would warrant manually overriding this.

**Comment by jeremiedimino** [comment]
I'm making this my before lunch time bugfix

**Comment by jeremiedimino** [comment]
Fixed in #3434. Note that the version constraint will only be inserted when using a version of the dune language >= 2.6
